provision/envoy.sh

```diff
@@ -1,13 +1,11 @@
 #!/bin/bash
 
-export GOPATH=/go/
-export CILIUM_USE_ENVOY=1
 export HOME_DIR=/home/vagrant
 export HOME=/home/vagrant
+export GOPATH="${HOME}/go"
```

Comment:
I'm not 100 % positive, but changes to `GOPATH` might affect the Ginkgo tests. Have you built this VM and ran the Ginkgo tests with it? If there is such a dependency, IMO we should try to decouple it separate from this PR so we don't have such tight coupling of the tests with the VM image being used.

Reply:
We can change this already, on the ginko tests we bump the VM version number with the required changes in the ginkgo tests

Reply:
OK, sounds good!